File: src/main/java/june_2025/MyLinkedList.java
package june_2025;

import java.util.ArrayList;
import java.util.List;

public class MyLinkedList {

    public static void main(String[] args) {
        MyLinkedList myLinkedList = new MyLinkedList();

        myLinkedList.deleteAtIndex(0);

//        myLinkedList.addAtHead(7);
//        myLinkedList.addAtHead(2);
//        myLinkedList.addAtHead(1);
//        myLinkedList.addAtIndex(3, 0);
//        myLinkedList.deleteAtIndex(2);
//        myLinkedList.addAtHead(6);
//        myLinkedList.addAtTail(4);
//
//        System.out.println(myLinkedList.get(4));
//        myLinkedList.addAtHead(4);
//        myLinkedList.addAtIndex(5, 0);
//        myLinkedList.addAtHead(6);
        System.out.println();
    }


    List<Integer> array;

    public MyLinkedList() {
        array = new ArrayList<>();
    }

    public int get(int index) {
        if (array.size() <= index) {
            return -1;
        }
        return array.get(index);
    }

    public void addAtHead(int val) {
        array.add(0, val);
    }

    public void addAtTail(int val) {
        array.add(array.size(), val);
    }

    public void addAtIndex(int index, int val) {
        if (array.size() < index) {
            return;
        }
        array.add(index, val);

    }

    public void deleteAtIndex(int index) {
        if (array.size() <= index) {
            return;
        }
        array.remove(index);
    }
}

File: python/second/my_linked_list.py
class MyLinkedList(object):

    def __init__(self):
        self.array = []

    def get(self, index):
        """
        :type index: int
        :rtype: int
        """
        if len(self.array) <= index:
            return -1
        return self.array[index]
        

    def addAtHead(self, val):
        """
        :type val: int
        :rtype: None
        """
        self.array.insert(0, val)
        

    def addAtTail(self, val):
        """
        :type val: int
        :rtype: None
        """
        self.array.append(val)
        

    def addAtIndex(self, index, val):
        """
        :type index: int
        :type val: int
        :rtype: None
        """
        if len(self.array) < index:
            return
        self.array.insert(index, val)
        

    def deleteAtIndex(self, index):
        """
        :type index: int
        :rtype: None
        """
        if len(self.array) <= index:
            return
        self.array.pop(index)
        
